io/net/htx_nic_devices.py

```diff
@@ -0,0 +1,591 @@
+# this script run IO stress on nic devices for give time.
+
+import os
+import re
+import time
+try:
+    import pxssh
+except ImportError:
+    from pexpect import pxssh
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils.process import CmdError
+
+
+class CommandFailed(Exception):
+    def __init__(self, command, output, exitcode):
+        self.command = command
+        self.output = output
+        self.exitcode = exitcode
+
+    def __str__(self):
+        return "Command '%s' exited with %d.\nOutput:\n%s" \
+               % (self.command, self.exitcode, self.output)
+
+
+class HtxNicTest(Test):
+
+    """
+    HTX [Hardware Test eXecutive] is a test tool suite. The goal of HTX is to
+    stress test the system by exercising all hardware components concurrently
+    in order to uncover any hardware design flaws and hardware hardware or
+    hardware-software interaction issues.
+    :see:https://github.com/open-power/HTX.git
+    :param mdt_file: mdt file used to trigger HTX
+    :params time_limit: how much time(hours) you want to run this stress.
+    :param host_ip: IP address of host
+    :param peer_ip: IP address of peer
+    :param peer_user: User name of Peer
+    :param peer_password: Password of Peer
+    :param host_intf1: Host N/W Interface 1 to run HTX on
+    :param host_intf2: Host N/W Interface 2 to run HTX on
+    :param peer_intf1: Peer N/W Interface 1 to run HTX on
+    :param peer_intf2: Peer N/W Interface 2 to run HTX on
+    :param net_id1: Net id 1 for first N/W Interface
+    :param net_id2: Net id 2 for second N/W Interface
+    """
+
+    def setUp(self):
+        """
+        Build 'HTX'.
+        """
+        if 'ppc64' not in process.system_output('uname -a', ignore_status=True,
+                                                shell=True, sudo=True):
+            self.cancel("Platform does not support HTX tests")
+
+        self.parameters()
+        self.login(self.peer_ip, self.peer_user, self.peer_password)
+        # Currently test assumes HTX is installed on both Host & Peer
+        # TODO: Clone HTX & build it
+        cmd = "test -d /usr/lpp/htx/"
+        try:
+            process.run(cmd, shell=True, sudo=True)
+        except CmdError:
+            self.cancel("HTX is not installed on Host")
+        try:
+            self.run_command(cmd)
+        except CommandFailed:
+            self.cancel("HTX is not installed on Peer")
+
+    def parameters(self):
+        self.host_ip = self.params.get("host_ip", '*', default=None)
+        self.peer_ip = self.params.get("peer_ip", '*', default=None)
+        self.peer_user = self.params.get("peer_user", '*', default=None)
+        self.peer_password = self.params.get("peer_password",
+                                             '*', default=None)
+        self.host_intf1 = self.params.get("host_intf1", '*', default=None)
+        self.host_intf2 = self.params.get("host_intf2", '*', default=None)
```

Comment:
Why are we assuming we will have 2 interfaces ? We can generalise this for 'n' interfaces, by getting a list of interfaces in the yaml file, as mentioned in the other comment.